src/projection.jl

```diff
@@ -122,6 +122,7 @@ ProjectTo(::Any) = identity
 
 # Thunks
 (project::ProjectTo)(dx::Thunk) = Thunk(project ∘ dx.f)
+(project::ProjectTo)(dx::InplaceableThunk) = Thunk(project ∘ dx.val.f)
```

Comment:
Would it be clearer to redispatch to the above?
```suggestion
(project::ProjectTo)(dx::InplaceableThunk) = project(dx.val)
```